pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractBaseDispatcher.java

```diff
@@ -93,22 +93,25 @@ protected AbstractBaseDispatcher(Subscription subscription, ServiceConfiguration
     public int filterEntriesForConsumer(List<? extends Entry> entries, EntryBatchSizes batchSizes,
             SendMessageInfo sendMessageInfo, EntryBatchIndexesAcks indexesAcks,
             ManagedCursor cursor, boolean isReplayRead, Consumer consumer) {
-        return filterEntriesForConsumer(Optional.empty(), 0, entries, batchSizes, sendMessageInfo, indexesAcks, cursor,
+        return filterEntriesForConsumer(null, 0, entries, batchSizes,
```

Comment:
The `null` value is LGTM, But if we introduce a static value like `MessageMetadata.Empty`, it will be more meaningful.

Reply:
`null` is good enough to represent an "empty" value. Especially we have `@Nullable` annotation here.